### PR TITLE
Add dangerous type blacklist feature to Akka.Serialization.Hyperion

### PR DESCRIPTION
**Pull request by Arkatufus**

### CODE REVIEW

File: docs/articles/networking/serialization.md [hunk 1]
```diff
@@ -304,6 +304,43 @@ akka {
 }
 ```
 
+## Danger of polymorphic serializer
+One of the danger of polymorphic serializers is the danger of unsafe object type injection into 
+the serialization-deserialization chain. Hyperion attempted to alleviate this by blacklisting
```
Comment (Aaronontheweb) [inline docs/articles/networking/serialization.md:309]: I'd add some caveats here:

1. This is an issue that primarily affects developers who allow third parties to pass messages directly to unsecured Akka.Remote endpoints, [a practice that we do not encourage](https://getakka.net/articles/remoting/security.html#akkaremote-with-virtual-private-networks) and do not see often in the wild.
2. This is an issue that also affects really, any other type of polymorphic serializer (i.e. JSON, BinaryFormatter, etc)
3. There are two approaches for solving this problem at point of the serializer: implement type filtering as shown here or implement schema-based serialization, which is more expensive to setup at first but fundamentally faster and more secure. Example: https://petabridge.com/cluster/lesson3

File: docs/articles/networking/serialization.md [hunk 2]
```diff
@@ -304,6 +304,43 @@ akka {
...
+- System.Diagnostics.Process
+- System.Management.IWbemClassObjectFreeThreaded
+
+If you needed to serialize one of these class, be warned that they can be used as an attack vector
+inside your system, you can turn off this feature using this inside your HOCON settings:
+```
+akka.actor.serialization-settings.hyperion.disallow-unsafe-type = false
+```
+
+> [!WARNING]
```
Comment (Aaronontheweb) [inline docs/articles/networking/serialization.md:339]: Ah ok good, you covered this here.

File: docs/articles/networking/serialization.md [hunk 3]
```diff
@@ -304,6 +304,43 @@ akka {
...
+- System.Windows.ResourceDictionary
+- System.Windows.Forms.BindingSource
+- Microsoft.Exchange.Management.SystemManager.WinForms.ExchangeSettingsProvider
+- System.Diagnostics.Process
+- System.Management.IWbemClassObjectFreeThreaded
+
+If you needed to serialize one of these class, be warned that they can be used as an attack vector
+inside your system, you can turn off this feature using this inside your HOCON settings:
+```
+akka.actor.serialization-settings.hyperion.disallow-unsafe-type = false
```
Comment (Aaronontheweb) [inline docs/articles/networking/serialization.md:336]: Need to mention, probably in an `[!IMPORTANT]` block that these security features are enabled by default as of Akka.NET v1.4.24

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Some feedback

File: docs/articles/networking/serialization.md (hunk 3 above)
Comment (Arkatufus) [reply docs/articles/networking/serialization.md:347]: Done

File: docs/articles/networking/serialization.md (hunk 1 above)
Comment (Arkatufus) [reply docs/articles/networking/serialization.md]: Used some of your wordings to improve the documentation.

File: src/contrib/serializers/Akka.Serialization.Hyperion/HyperionSerializerSetup.cs [hunk 4]
```diff
@@ -88,14 +99,21 @@ public class HyperionSerializerSetup : Setup
         /// into a much simpler serialized objects.
         /// </summary>
         public IEnumerable<Surrogate> Surrogates { get; }
+        
+        /// <summary>
+        /// If set, will cause the Hyperion serializer to block potentially dangerous and unsafe types
+        /// from being deserialized during run-time
```
[the remark this answers is not in the document]
Comment (Arkatufus) [reply src/contrib/serializers/Akka.Serialization.Hyperion/HyperionSerializerSetup.cs]: Done.

File: src/contrib/serializers/Akka.Serialization.Hyperion/reference.conf [hunk 5]
```diff
@@ -14,11 +14,35 @@
 			# between actor systems using different assembly versions, making incremental 
 			# cluster updates less restrictive.
 			version-tolerance = true
+			
+			# If true, hyperion serializer will block potentially dangerous and unsafe types
+			# from being deserialized during run-time. The types that are blocked are:
```
[the remark this answers is not in the document]
Comment (Arkatufus) [reply src/contrib/serializers/Akka.Serialization.Hyperion/reference.conf]: Removed the overly verbose list of classes and points to the documentation instead.

**Review by Aaronontheweb — APPROVED** [review]
LGTM - nice work. We'll update this again once we cover this for Newtonsoft.Json in a future release.

**Pull request merged**